packages/zoltan2/test/helpers/AdapterForTests.hpp

```diff
@@ -92,25 +92,21 @@ using std::string;
 using namespace Zoltan2_TestingFramework;
 
 // helper struct to store both an adapter and the coordinate adapter
-struct AdapterWithOptionalCoordinateAdapter
+struct AdapterWithTemplateName
 {
-  Zoltan2_TestingFramework::base_adapter_t * mainAdapter = nullptr;
-  Zoltan2::VectorAdapter<tMVector_t> * coordinateAdapter = nullptr;  // may be NULL if coordinates are not available
+  Zoltan2::BaseAdapterRoot * adapter = nullptr; // generic base class
```

Comment:
Here, we store the adapter as a pointer to BaseAdapterRoot, the lowest base class, right?  And EAdapterType says which type of adapter it is, so that a switch can cast back to the original adapter type, right?  But I think there are still casts to basic_id_t elsewhere.  Are the casts to basic_id_t still needed elsewhere?  Should they be casts to the original adapter type?  (I admit I never quite understood the casts to basic_id_t before; it seemed that rather than casting, the functions that used the adapters should be templated on adapter type.  I never had time to thoroughly understand the issue, though.)